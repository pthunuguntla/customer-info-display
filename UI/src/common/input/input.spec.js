
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from './';

describe('<Button /> with no props', () => {
    it('renders correctly ', () => {
        const wrapper = shallow(
            <Input
                placeholder={"Type Something"}
                searchValue={"test"}
                onInputChange={jest.fn().mockImplementation(() => { return 'test' })}
            />);
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should return test after executing  onInputChange', () => {

        const onInputChangeMock = jest.fn().mockImplementation(() => { return 'test' });

        const wrapper = mount(
            <Input
                placeholder={"Type Something"}
                searchValue={"test"}
                onInputChange={onInputChangeMock}
            />);

        wrapper.find('input').simulate('change');
        expect(onInputChangeMock).toBeCalledWith('test');
    });
});