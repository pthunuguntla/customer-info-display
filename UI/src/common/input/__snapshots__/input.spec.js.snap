// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> with no props renders correctly  1`] = `
<input
  className="input"
  onChange={[Function]}
  placeholder="Type Something"
  type="text"
  value="test"
/>
`;
